go-controller/pkg/util/net.go

```diff
@@ -34,13 +34,17 @@ func intToIP(i *big.Int) net.IP {
 // IPToUint32 returns a uint32 of an IPv4/IPv6 string
 func IPToUint32(egressIP string) uint32 {
 	ip := net.ParseIP(egressIP)
+	if ip == nil {
+		return 0
+	}
 	if utilnet.IsIPv6(ip) {
 		// This can obviously not be done for IPv6. But the logic here is:
 		// "allow users to create IPv6 egress IP addresses with a 1/(2^32)
 		// probability that they might collide. Or just use shared gateway
 		// mode, and live without this risk."
 		return binary.BigEndian.Uint32(ip[12:16])
```

Comment:
This should probably share code with `IPAddrToHWAddr`, which has the same problems of needing to call `To4` on IPv4 addresses, and needing to convert IPv6 addresses to only 4 bytes (which it does by hashing them, because we had problems with collisions when we just truncated them).